incubator/hnc/pkg/reconcilers/hnc_config.go

```diff
@@ -222,7 +221,14 @@ func (r *ConfigReconciler) syncObjectReconcilers(ctx context.Context, inst *api.
 // syncActiveReconcilers syncs object reconcilers for types that are in the Spec. If an object reconciler exists, it sets
 // its mode according to the Spec; otherwise, it creates the object reconciler.
 func (r *ConfigReconciler) syncActiveReconcilers(ctx context.Context, inst *api.HNCConfiguration) error {
+	// exist keeps track of existing types in the `config` singleton.
+	exist := map[schema.GroupVersionKind]bool{}
 	for _, t := range inst.Spec.Types {
+		// If there are multiple configurations of the same type, we will follow the first
+		// configuration and ignore the rest.
+		if r.isTypeConfigured(inst, t, exist) {
```

Comment:
Nit: this doesn't _just_ return an answer, it both checks for something invalid happening, and _sets_ a condition. Maybe `r.ensureNoDuplicateTypeConfigurations`?

Reply:
Thanks for the suggestion! Changed to `r.ensureNoDuplicateTypeConfigurations`.